components/tikv_util/src/buffer_vec.rs

```diff
@@ -350,7 +350,7 @@ impl<'a> Iterator for Iter<'a> {
     type Item = &'a [u8];
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.nth(0)
+        Self::nth(self, 0)
```

Comment:
Curious why Clippy wants this change

Reply:
Since Clippy checks for the use of iter.nth(0), see https://rust-lang.github.io/rust-clippy/master/index.html#iter_nth_zero

I guess it is not expected behavior in the implementation of an Iterator.

Reply:
I see, thanks! :+1: